components/server/src/env.ts

```diff
@@ -47,7 +47,13 @@ export class Env extends AbstractComponentEnv {
 
     readonly previewFeatureFlags: NamedWorkspaceFeatureFlag[] = (() => {
         const v = process.env.EXPERIMENTAL_FEATURE_FLAGS;
-        return !!v ? JSON.parse(v) : [];
+        if (!v) return [];
+        try {
+            const flags = JSON.parse(v);
+            return Array.isArray(flags) ? flags : [];
+        } catch (_) {
+            return [];
```

Comment:
that means, any misconfiguration will be ignored.

🅰️ log the error, or
🅱️ let it fail

Reply:
Maybe let it fail then.

Should we also throw when the parsed value is not an Array?

Reply:
sounds good!